### ISSUE TITLE
results different c-python vs transcrypt

### ISSUE THREAD

**Issue by jonathan-arrender-smith** [QQuick/Transcrypt#264]
Run the following in python 3.5 and transcrypt. The results are quite different.

```python3
from typing import Sequence

def subsum(total: int, items: Sequence[int]):
    """Compute all sublists of items that sum to total.

    >>> subsum(10, range(1, 10))
    [[4, 6], [3, 7], [2, 8], [2, 3, 5], [1, 9], [1, 4, 5], [1, 3, 6], [1, 2, 7], [1, 2, 3, 4]]

    """
    if total == 0:
        return [[]]
    elif len(items) == 0:
        return []
    else:
        first, others = items[0], items[1:]
        results = subsum(total, others)
        if total >= first:
            results += [[first] + ls for ls in subsum(total - first, others)]
        return results

def subsum_main():
    results = subsum(10, range(1, 10))
    print('start={}'.format(len(results)))
    print()
    for s in results:
        print("len={}".format(len(s)))
        for i in s:
            print(i)
        print('-----')

subsum_main()
```

**Comment by JdeH** [comment]
Thanks!

We'll look into this a.s.a.p.

**Comment by neriusmika** [comment]
The problem is with list concatenation. You need to use `extend()` or opov pragma in transcrypt. Just change your code to:

```python
        ...
        if total >= first:
            __pragma__('opov')
            results.extend([[first] + ls for ls in subsum(total - first, others)])
            __pragma__('noopov')
        return results
```

**Comment by neriusmika** [comment]
By the way, even if you use opov, += operator doesn't work correctly. I think that `Array.prototype.__add__` must be fixed or `Array.prototype.__iadd__` added.

**Comment by JdeH** [comment]
@neriusmika 
+= and friends are not yet part of Transcrypt, since I considered them secundary, offering only slight notational convenience. Since &, ^ and / have now also been added, I'll add += etc. as well. I'll make __builtin__ slightly larger, but as more JS5 workarounds will be replaced by JS6 code, it'll shrink again. 

@jonathan-arrender-smith 
Indeed for efficiency reasons operator overloading is locally optional (http://sterlicht.alwaysdata.net/transcrypt.org/docs/html/special_facilities.html#operator-overloading-pragma-opov-and-pragma-noopov). If you want it global use the -o / --opov command line switch (http://sterlicht.alwaysdata.net/transcrypt.org/docs/html/special_facilities.html#operator-overloading-pragma-opov-and-pragma-noopov), but I think @neriusmika' s solution is better. Let me know if it works for you.

I'll relabel this issue as an enhancement, referring to the planned additon of += etc.

**Comment by JdeH** [comment]
Resolved and commited, see #283
